mrmustard/physics/ansatze.py

```diff
@@ -417,8 +417,25 @@ def __add__(self, other: PolyExpBase) -> PolyExpBase:
         Adds two ansatze together. This means concatenating them in the batch dimension.
         In the case where c is a polynomial of different shapes it will add padding zeros to make
         the shapes fit. Example: If the shape of c1 is (1,3,4,5) and the shape of c2 is (1,5,4,3) then the
-        shape of the combined object will be (2,5,4,5).
+        shape of the combined object will be (2,5,4,5). It also pads A and b, to account for an eventual
+        different number of polynomial wires.
         """
+        (_, n, _) = self.mat.shape
+        (_, m, _) = other.mat.shape
+        self_num_poly, _ = self.polynomial_shape
+        other_num_poly, _ = other.polynomial_shape
+        if self_num_poly - other_num_poly != n - m:
+            raise ValueError("Inconsistent polynomial shapes.")
+        if n < m:
+            self.mat = math.pad(self.mat, ((0, 0), (0, m - n), (0, m - n)))
```

Comment:
I believe these are mutating `self` and `other` (below). Is this what we want? I would be a bit worried about having unintended effects in this case. Maybe we should use similar logic for how we handle `c` instead i.e. a0_new, a1_new

Reply:
Also, are we able to combine this logic with the `c` padding logic below in any way? That should clean some things up

Reply:
Hm, yes maybe. Thanks for the suggestion, I'll see what I can do. 👍